src/js/components/Header/Header.js

```diff
@@ -1,19 +1,141 @@
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, background, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const containerRef = useForwardedRef(ref);
+  const [stickyStyles, setStickyStyles] = useState();
+  const { ...pageContext } = useContext(PageContext);
+
+  useEffect(() => {
+    // threshold inital value set to 0
+    const threshold = 0;
+    // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    // run our event listener callback once in each requestAnimationFrame
+    let runEventListener = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        runEventListener = false;
+        return;
+      }
+      if (scrollY === 0) {
+        runEventListener = false;
+        setStickyStyles(undefined);
+      }
+      if (scrollY < lastScrollY) {
+        setStickyStyles({
+          width: `${containerRef.current.getBoundingClientRect().width}px`,
+          height: `${containerRef.current.getBoundingClientRect().height}px`,
+          left: `${containerRef.current.getBoundingClientRect().left}px`,
+          right: `${containerRef.current.getBoundingClientRect().right}px`,
+          zIndex: `${theme.header?.sticky?.zIndex}`,
+          position: 'fixed',
+          transition: 'top 0.6s',
+          top: '0',
+        });
+      } else
+        setStickyStyles({
+          top: `-${containerRef.current.getBoundingClientRect().height}px`,
+        });
+      lastScrollY = scrollY > 0 ? scrollY : 0;
+      runEventListener = false;
+    };
+
+    const onScroll = () => {
+      if (!runEventListener) {
+        // calculate the new offset after the page gets rendered
+        window.requestAnimationFrame(updateScrollDir);
+        runEventListener = true;
+      }
+    };
+
+    window.addEventListener('scroll', onScroll);
+
+    return () => window.removeEventListener('scroll', onScroll);
+  }, [containerRef, stickyStyles, theme.header?.sticky?.zIndex]);
+
+  if (sticky?.type === 'scrollup') {
+    if (sticky?.fill) {
+      return (
+        <>
+          <Box
+            width={stickyStyles && stickyStyles.width}
+            height={stickyStyles && stickyStyles.height}
+          />
+          <Box
+            align="center"
+            as="header"
+            height={stickyStyles && stickyStyles.height}
+            width={stickyStyles && stickyStyles.width}
+            background={background}
+            style={stickyStyles}
+            {...rest}
+            ref={containerRef}
+          >
+            <Box
+              align="center"
+              direction="row"
+              fill
```

Comment:
```suggestion
              fill="horizontal"
```

Reply:
Would it just be cleaner to import `PageContent` and have this be:

<PageContent {...defaultHeaderProps} {...rest} />

Reply:
Feels like we're repeating logic and work that PageContent already knows how to do

Reply:
that's a possibility if we like this direction I can clean that up